### DISCUSSION TITLE
Serving UI using a base path

### DISCUSSION THREAD

**Discussion by freudl** [inovex/scrumlr.io#4092] [Q&A] [answered]
Hi,

like your project but I'm not able to configure it to serve the UI component from a base path, e.g. `https://example.com/retro`.

I managed to show `index.html` by updating the [environment](https://github.com/inovex/scrumlr.io/blob/main/.env.development#L6). 
Once I login anonymously the [redirect points to `/new`](https://github.com/inovex/scrumlr.io/blob/main/src/routes/LoginBoard/LoginBoard.tsx#L32) instead of `/retro/new`. Displaying the [legal notes](https://github.com/inovex/scrumlr.io/blob/main/src/routes/Router.tsx#L28) from the footer are not working.

As far as I understood, [`BrowserRouter`](https://github.com/inovex/scrumlr.io/blob/main/src/routes/Router.tsx#L24) might need to use [`basename`](https://reactrouter.com/en/main/routers/create-browser-router#basename), but this seems to break other features, e.g. login.

* Am I on the right track?
* Does the reverse proxy based approach others [suggested](https://github.com/inovex/scrumlr.io/issues/274) actually work?

**Comment by Schwehn42** [comment]
hi Stephan, thanks for the question.

@brandstetterm could you give some pointers for this when you have time?

**Reply by brandstetterm** [reply]
Hey, I'm going to have a look on it in the coming week :)

**Comment by brandstetterm** [comment, marked as answer]
Hey @freudl,
thanks for reaching out and sorry for the late response. You're definitely on the right track with setting the `basename` of our `BrowserRouter`. Since our landing page is a separate project, we had to use some `<a href="/">` instead of the `<Link>` component. Therefore, you have to update some the hrefs & one `window.location.pathname` to include your new basename in the components BoardHeader, LoginBoard, NewBoard and Legal.

Also, the problem with our login is the following: If a unauthenticated client tries to access either a board or the page to create a new board, he will be redirected to our login page. After the client logged in, it will be redirected to the page visited before. For some weird reason, `location.pathname` doesn't include the basename if invoked within our code (it works as intended from the console), so it redirects back to `/new` instead of `/<basename>/new`.
